mypy/nodes.py

```diff
@@ -2234,21 +2234,55 @@ def __getattribute__(self, attr: str) -> None:
 
 
 class SymbolTableNode:
+    """Description of a name binding in a symbol table.
+
+    These are only used as values in module (global), function (local)
+    and class symbol tables (see SymbolTable). The name that is bound is
+    the key in SymbolTable.
+
+    Symbol tables don't contain direct references to AST nodes primarily
+    because there can be multiple symbol table references to a single
+    AST node (due to imports and aliases), and different references can
+    behave differently. This class describes the unique properties of
+    each reference.
+
+    The most fundamental attributes are 'kind' and 'node'.  The 'node'
+    attribute defines the AST node that the name refers to.
+
+    For many bindings, including those targeting variables, functions
```

Comment:
Maybe it will be helpful to note that this is a bit similar to how CPython symbol tables work. There are no ``.node``, only ``.kind`` (and there are more than three kinds however). These kinds are used to emit correct lookup opcodes, so that the right ``.node`` will be found by the interpreter at runtime (I know this comparison is vague).

Reply:
Hmm talking about CPython implementation details might not be helpful, as many readers likely aren't familiar with them.